### DISCUSSION TITLE
[Feature]: wasm

### DISCUSSION THREAD

**Discussion by aticzz** [fabricjs/fabric.js#10093] [General]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have searched and referenced existing issues, feature requests and discussions
- [X] I am filing a **FEATURE** request.

### Description

Web assembly wasm

### Current State

I would like to request the addition of WebAssembly (WASM) support to Fabric.js to enhance the library's performance, particularly in CPU-intensive operations such as image manipulation, complex object rendering, and large-scale canvas interactions.

### Additional Context
My entire application relies heavily on Fabric.js for several computationally intensive tasks:

Image Tint Filters: I frequently apply tint filters to images, which can be resource-intensive, especially when dealing with high-resolution images or multiple layers.

Text Fitting with Recursive Functions: The application uses recursive functions to precisely fit text within a reference layer, a process that demands efficient execution to maintain design accuracy.

Video Creation from Canvas: I generate videos directly from the canvas, necessitating smooth rendering and efficient processing to ensure the final video is of high quality.

Multiple Canvases in the Same DOM: My application also involves managing multiple canvases within the same DOM. Each canvas operates independently.

Canvas JSON Saving: I save the JSON representation of the canvas, which is critical for persisting and reloading complex states and designs.

Given the complexity and performance demands of these tasks, integrating WebAssembly (WASM) into Fabric.js would be extremely beneficial. WASM's near-native execution speed would significantly reduce processing times and enhance the overall performance, especially in scenarios involving multiple canvases and complex operations.

_No response_

**Comment by asturur** [comment]
Filters and drawing are already native executing speed, the JS around them is small.
Do you have any flamegraph that shows that you are spending time in js userland more than native canvas api from the browser?
